static/app/components/events/interfaces/spans/types.tsx

```diff
@@ -90,3 +90,31 @@ export type OrphanTreeDepth = {
 };
 
 export type TreeDepthType = SpanTreeDepth | OrphanTreeDepth;
+
+export type IndexedFusedSpan = {
+  span: RawSpanType;
+  indexed: string[];
+  tagKeys: string[];
+  tagValues: string[];
+  dataKeys: string[];
+  dataValues: string[];
+};
+
+export type FuseResult = {
+  item: IndexedFusedSpan;
+  score: number;
+};
+
+export type FilterSpans = {
+  results: FuseResult[];
+  spanIDs: Set<string>;
+};
+
+export type SpanFuseOptions = {
+  keys: ('indexed' | 'tagKeys' | 'tagValues' | 'dataKeys' | 'dataValues')[];
```

Comment:
nit: Can we split out and name the type of the key here? something like

```typescript
type FuseKey = 'indexed' | 'tagKeys' | 'tagValues' | 'dataKeys' | 'dataValues';
```

Reply:
Done in https://github.com/getsentry/sentry/pull/26544/commits/dcba31d451240674669d712227a4420ec1516d05